crates/ruff_python_ast/src/nodes.rs

```diff
@@ -767,12 +767,62 @@ impl From<ExprIf> for Expr {
     }
 }
 
+#[derive(Debug, Clone, PartialEq)]
+pub struct DictItem {
+    pub key: Option<Expr>,
+    pub value: Expr,
+}
+
+impl DictItem {
+    fn key(&self) -> Option<&Expr> {
+        self.key.as_ref()
+    }
+
+    fn value(&self) -> &Expr {
+        &self.value
+    }
+}
+
+impl Ranged for DictItem {
+    fn range(&self) -> TextRange {
+        TextRange::new(
+            self.key.as_ref().map_or(self.value.start(), Ranged::start),
+            self.value.end(),
+        )
+    }
+}
+
 /// See also [Dict](https://docs.python.org/3/library/ast.html#ast.Dict)
 #[derive(Clone, Debug, PartialEq)]
 pub struct ExprDict {
     pub range: TextRange,
-    pub keys: Vec<Option<Expr>>,
-    pub values: Vec<Expr>,
+    pub items: Vec<DictItem>,
+}
+
+impl ExprDict {
+    pub fn iter_keys(&self) -> DictKeyIterator {
+        DictKeyIterator::new(&self.items)
+    }
+
+    pub fn iter_values(&self) -> DictValueIterator {
+        DictValueIterator::new(&self.items)
+    }
+
+    pub fn iter_items(&self) -> Iter<'_, DictItem> {
+        self.items.iter()
+    }
+
```

Comment:
I recommend returning a slice here. It's often easier to work with a slice than an iterator.

```suggestion
    pub fn iter_items(&self) -> &[DictItem] {
        self.items.iter()
    }

```

Reply:
I feel like I'd either prefer to keep the returned type consistent with `iter_key()` and `iter_values()`, or just remove this method entirely. If you want a `[DictItem]` slice, you can just access the `.items` field on the `ExprDict` node directly; it's a public field

Reply:
nit: in that case, I'd remove the `iter_items` method.

Reply:
> I feel like I'd either prefer to keep the returned type consistent with iter_key() and iter_values(), or just remove this method entirely. If you want a [DictItem] slice, you can just access the .items field on the ExprDict node directly; it's a public field

That's true. The reason why I often prefer methods is because it avoids the sometimes awkward `&` or `as_ref()`

```rust
for item in &dict.items
```

vs 

```
for item in dict.items() 
```